src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -101,6 +102,12 @@ public class QueryProcessor implements QueryHandler
     // counters. Callers of processStatement are responsible for correctly notifying metrics
     public static final CQLMetrics metrics = new CQLMetrics();
 
+    // Paging size to use when preloading prepared statements.
+    public static final int PRELOAD_PREPARED_STATEMENTS_FETCH_SIZE = 5000;
+
+    // Maximum number of prepared statements to preload from prepared statement cache.
+    public static final int PRELOAD_PREPARED_STATEMENTS_MAX = 1_000_000;
```

Comment:
does 1 million feel about right here?  Would there be any value in making this configurable? I can't imagine anyone would actually want to do this, but we could make it a system property if that has some value.

Reply:
Thinking about it we don't need to come up with this new limit. We already have a limit on the cache size in MB. Once you hit that it makes no sense to just be trashing the cache. So just drop this and monitor cache instead.

Reply:
:+1: (further discussion [here](https://github.com/apache/cassandra/pull/3917#discussion_r2042139638))